### PR TITLE
Added ability to set custom cache system.

### PR DESCRIPTION
**Pull request by garygreen**
Useful if you want to have a custom way of storing the cache.

### CODE REVIEW

File: Image.php [hunk 1]
```diff
@@ -104,11 +104,36 @@ class Image
     protected $cache;
```
Comment (Gregwar) [inline Image.php]: Looks like the cache attribute is not defaulted to null
It would be a better practice since you test that on L113

Comment (Gregwar) [inline Image.php]: After looking the docs, looks like it doesnt really matter actually

**Comment by Gregwar** [comment]
I'd prefer reviewing the interface pr first and making an update with the correct type hinting

**Comment by garygreen** [comment]
@Gregwar I've updated the PR to type hint the `CacheInterface` since it's been merged. 

Really keen to get this PR merged, as it I would like to change the default caching prefix size `setPrefixSize` -- idea is I'll create a singleton Cache instance that will be used across all Gregwar instances.

**Comment by garygreen** [comment]
Ping @Gregwar . Any news?

**Comment by Gregwar** [comment]
Sorry for the delays, it seems good to me

**Pull request merged**